src/bsoncxx/types.hpp

```diff
@@ -530,6 +556,11 @@ struct BSONCXX_API b_codewscope {
         : code(std::forward<T>(code)), scope(std::forward<U>(scope)) {
     }
 
+    ///
+    /// Default constructor for b_codewscope.
+    ///
+    BSONCXX_INLINE b_codewscope() = default;
+
```

Comment:
This is the only one that leaves me a little cold. The default constructor for this is going to leave you with a string_view pointing to nothing (not the empty string!), but a view that will view the default empty document. That seems a little weird. I think it is a side effect of our not allowing views to not view something. Is that something we should consider changing.

Also, will you please fix a nit for me, that I was reminded of while thinking about this?

In `view.cpp`, the `k_default_view` object is not const! Could you please fix that?